optuna/integration/mlflow.py

```diff
@@ -234,12 +247,8 @@ def _set_tags(self, trial: optuna.trial.FrozenTrial, study: optuna.study.Study)
         directions = [d.name for d in study.directions]
         tags["direction"] = directions if len(directions) != 1 else directions[0]
 
-        tags.update(trial.user_attrs)
         distributions = {(k + "_distribution"): str(v) for (k, v) in trial.distributions.items()}
-        tags.update(distributions)
-
-        if self._tag_study_user_attrs:
-            tags.update(study.user_attrs)
+        tags.update({**distributions, **trial.user_attrs, **study.user_attrs})
```

Comment:
In my understanding, if `trial.user_attrs` and `study.user_attrs` have the same key, i.e., `contributors` in the following example, `study.user_attrs`'s value overwrites `trial.user_attrs`'s value, `"contributors": "mamu"`, in MLflow's logging procedure.


```python
def objective(trial):
    x = trial.suggest_float("x", -10, 10)
    trial.set_user_attr("contributors", ["akiba", "sano"])
    trial.study.set_user_attr("contributors", "mamu")
    return (x - 2) ** 2
```

I think there are a few ways to avoid this

- just adding a note section to the documentation because I think this is unlikely to happen in practice. 
- adding a warning message in this callback when the same key appears in `trial.user_attr` and `study.user_attr`
- adding prefix to each key: `trial.user_attr. contributors` and `study.user_attr. contributors` to make sure that keys are unique.

cc: @toshihikoyanase

Reply:
Thank you for spotting this! I think we could prefix colliding keys to avoid this, since it will not complicate logic by too much.

Reply:
Indeed. I agree with you!

Reply:
Awesome! Patched in 40fafab38f0b9487d05dadc8cda7cc5ee8eaa293.